File: 프로그래머스/lv0/120869. 외계어 사전/외계어 사전.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<string> spell, vector<string> dic)
{
    string arr3 = "";
    sort(spell.begin(), spell.end());

    for (int i = 0; i < dic.size(); i++)
        sort(dic[i].begin(), dic[i].end());

    for (int i = 0; i < spell.size(); i++)
        arr3.append(spell[i]);
    
    for (int i = 0; i < dic.size(); i++)
    {
        if (dic[i] == arr3)
            return 1;
    }
          
    return 2;
}

File: 프로그래머스/0/120869. 외계어 사전/외계어 사전.java
import java.util.Arrays;

class Solution {

    public int solution(String[] spell, String[] dic) {
        Arrays.sort(spell);
        String join = String.join("", spell);
        for (String s : dic) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String convert = new String(charArray);
            if (join.equals(convert)) {
                return 1;
            }
        }
        return 2;
    }
}